lib/src/kvs/tx.rs

```diff
@@ -37,42 +37,8 @@ const LOG: &str = "surrealdb::txn";
 /// A set of undoable updates and requests against a dataset.
 #[allow(dead_code)]
 pub struct Transaction {
-	pub(super) inner: Inner,
-	pub(super) cache: Cache,
-}
-
-#[allow(clippy::large_enum_variant)]
-pub(super) enum Inner {
-	#[cfg(feature = "kv-mem")]
-	Mem(super::mem::Transaction),
-	#[cfg(feature = "kv-rocksdb")]
-	RocksDB(super::rocksdb::Transaction),
-	#[cfg(feature = "kv-indxdb")]
-	IndxDB(super::indxdb::Transaction),
-	#[cfg(feature = "kv-tikv")]
-	TiKV(super::tikv::Transaction),
-	#[cfg(feature = "kv-fdb")]
-	FDB(super::fdb::Transaction),
-}
-
-impl fmt::Display for Transaction {
-	fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
-		#![allow(unused_variables)]
-		match &self.inner {
-			#[cfg(feature = "kv-mem")]
-			Inner::Mem(_) => write!(f, "memory"),
-			#[cfg(feature = "kv-rocksdb")]
-			Inner::RocksDB(_) => write!(f, "rocksdb"),
-			#[cfg(feature = "kv-indxdb")]
-			Inner::IndxDB(_) => write!(f, "indexdb"),
-			#[cfg(feature = "kv-tikv")]
-			Inner::TiKV(_) => write!(f, "tikv"),
-			#[cfg(feature = "kv-fdb")]
-			Inner::FDB(_) => write!(f, "fdb"),
-			#[allow(unreachable_patterns)]
-			_ => unreachable!(),
-		}
-	}
+	pub(super) inner: Box<dyn TransactionFacade + Send + Sync>,
```

Comment:
I'm not sure if we decided this based on any numbers but, I was under the impression that we avoided dynamic dispatching for performance reasons.

Did you take that into consideration by any chance @stevefan1999-personal? I'm just asking so that we may have a better foundation for further discussion!

@rushmorem Do you have any insight on that?

Reply:
Well if you use an enum and a match statement to dispatch to call to different places, I think it is not better than dynamic dispatch because the match arms/branches caused a lot of indirection anyway, but even if we assuned it can do better (does branch prediction really matter here?) I'd rather focusing on the boilerplate side of thing and long term maintainance.

That I don't want to take so much time to carefully check all the potential places I needed to insert the missing match arm to just use my new data store implementation. 

If adding a bunch of match arms for performance sake worth the sacrifice of developer experience, we can obvious get a compromise if we remove the generic match arm on dev builds, though. 

Then the compiler will error-hint you where you have missed a match exhausion. We can use a production feature switch to enable the generic match arm.

I will show a code example later but I'm quite tired right now.

Reply:
@stevefan1999-personal Thank you so much for your reply and patience here!

Just confirmed that for this specific topic, we are not strictly settled on either way (dyn dispatching vs branching).

> That I don't want to take so much time to carefully check all the potential places I needed to insert the missing match arm to just use my new data store implementation.

I can see that. Actually, that's basically what happened to me when I contributed FDB backend.

> I will show a code example later but I'm quite tired right now.

Please do not feel like you have to spend more and more time on this! I believe that you spend a lot on this already and that's us who has the ball right now. Of course, it's absolutely fine you are doing this for 100% fun but I also understand doing OSS contributions doesn't go that simple sometimes...

I'm saying this because I just realized we forced you to wait for 4 months until the decision is made. It's okay if I overthought it, but you're feeling tired due to this, I'm very sorry about that. We are a small team with limited bandwidth but I don't think it can justify the radio silence...

Will try to get back to you soon.

Reply:
> but if you're feeling tired due to this, I'm very sorry about that

I think you overinterpreted that because I mean I'm literally lying on my bed because my stomach is not good at that moment😂There's no need to apologize🙏I know I'm not a clever person to effectively express myself after all.

I'm not sure if there's a Clippy rule that would facilitate or supports what I want to express. Given something like:
```rust
enum Actions {
    Foo,
    Bar,
    Baz
}

match action {
    #[cfg(feature = "foo")]
    Actions::Foo => println!("foo"),
    #[cfg(feature = "bar")]
    Actions::Bar => println!("bar"),
    _ => unreachable!()
}
```

And then Clippy would **warn** me "`Actions::Baz` is missing on this `match action` expression and will fall though to the wildcard match, are you sure that's the intended behavior?", something like that.